github.go

```diff
@@ -69,30 +70,32 @@ func NewGithubClient(
 	connectionTimeout time.Duration,
 ) *GithubLocator {
 	var httpClient *http.Client
+	ctx, cancel := ensureTimeout(ctx, connectionTimeout)
```

Comment:
if there is more than 10s from the client creation to the first request - it will already be cancelled